api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -472,7 +476,39 @@ public void purgeProgram_shouldPurgeProgramWithPatientsEnrolled() {
 		// make sure that the program was deleted properly
 		assertNull(Context.getProgramWorkflowService().getProgram(2));
 	}
-	
+	@Test
+	public void shouldTestGetAllProgramAttributeTypes() throws Exception {
+	assertEquals(1, pws.getAllProgramAttributeTypes().size());
+	}
+
+	@Test
+	public void shouldTestGetProgramAttributeType() throws Exception {
+
+		ProgramAttributeType programAttributeType  = pws.getProgramAttributeType(1);
+		assertEquals("d7477c21-bfc3-4922-9591-e89d8b9c8efb",programAttributeType.getUuid());
+	}
+
+	@Test
+	public void shouldTestGetProgramAttributeTypeByUuid() throws Exception {
+		ProgramAttributeType p = pws.getProgramAttributeTypeByUuid("d7477c21-bfc3-4922-9591-e89d8b9c8efb");
+		assertEquals("ProgramId",p.getName());
+	}
+
+	@Test
+	public void shouldTestSaveProgramAttributeType() throws Exception {
+		assertEquals(1,pws.getAllProgramAttributeTypes().size());
+		ProgramAttributeType programAttributeType = new ProgramAttributeType();
+		programAttributeType.setName("test");
+		pws.saveProgramAttributeType(programAttributeType);
+		assertEquals(2,pws.getAllProgramAttributeTypes().size());
+	}
+
+	@Test
+	public void shouldTestPurgeProgramAttributeType() throws Exception {
+		ProgramAttributeType programAttributeType = pws.getProgramAttributeType(1);
+		pws.purgeProgramAttributeType(programAttributeType);
+		assertEquals(0,pws.getAllProgramAttributeTypes().size());
```

Comment:
Would be clearer if you started by asserting that the size was more than the one after purging.

Reply:
Okay, I'm not that conversant, just copy pasting the code.

Reply:
that would fail since after purging, size should be 0

Reply:
Sorry i mean't to say "was more by one".  That is, "by" instead of "than"

Reply:
This test is supposed to check if purging works, right? we have 1 record in the database, after purging and then fetch, the results should 0 which the test has accomplised, if the record wasn't there it would fail on fetch since we are fetching by id `ProgramAttributeType programAttributeType = pws.getProgramAttributeType(1);`

Reply:
Actually our common practice is to shield the test from changes to the number of records in the test dataset file. If some one is working on another aspect and adds more records to the test dataset file, this test will start failing. So we recommend something like getAllRecordsValue and then after purging, you assert that the size is now (getAllRecordsValue - 1)